test/support/test_config.h

```diff
@@ -117,7 +117,8 @@
 #endif
 #endif //!defined(_ENABLE_RANGES_TESTING)
 
-#define _ENABLE_STD_RANGES_TESTING _ONEDPL_CPP20_RANGES_PRESENT
+#define _ENABLE_STD_RANGES_TESTING ONEDPL_HAS_RANGES_API

```

Comment:
I guess oneDPL should be included to guarantee that `ONEDPL_HAS_RANGES_API` is available.

Reply:
At least each test includes `execution` , which includes `onedpl_config.h`

Reply:
This is dangerous. If you want to leave this as is, add a corresponding comment. 
This is also not ideal, but you may consider defining `_ENABLE_STD_RANGES_TESTING` in `std_ranges_tests.h`, which includes `test_config` and oneDPL in the desired order.

Reply:
> This is dangerous

why?
std_ranges_test.h:

```
#include <oneapi/dpl/execution>

#include "support/test_config.h"
...
```

Reply:
Also, we can add a kind of preprocessor error when _ENABLE_STD_RANGES_TESTING is not defined at all.

Reply:
> why?

Whether will `_ENABLE_STD_RANGES_TESTING` be properly defined or not depends on a header inclusion order in another file. Somebody can change it without knowing about such a dependency. That's why I recommended defining  `_ENABLE_STD_RANGES_TESTING` right in `std_ranges_test.h:` so that the dependency presence is obvious.

Reply:
> Also, we can add a kind of preprocessor error when _ENABLE_STD_RANGES_TESTING is not defined at all.

Sounds reasonable either.

Reply:
1) For std_ranges_* we have strongly "rules"  - one of them to include "std_ranges_test.h".
2) " kind of preprocessor error"... but where? in each test? it is an overkill... In "std_ranges_test.h"? it doesn't make sense if developer doesn't include "std_ranges_test.h"....
I suggest keep as is. I don't see problem here, even _ENABLE_STD_RANGES_TESTING  is not defined at all - we will see "skipped" in the test log.